DuckDB.NET.Data/ConnectionString/DuckDBConnectionStringParser.cs

```diff
@@ -39,9 +40,13 @@ public static DuckDBConnectionString Parse(string connectionString)
 
     private static string GetDataSource(IReadOnlyDictionary<string, string> properties)
     {
+        var lowerCaseProperties = properties.ToDictionary(kv => kv.Value.ToLower(CultureInfo.InvariantCulture), kv => kv.Value);
```

Comment:
There is no need to create a second dictionary, the best option is to make `properties` case-insensitive by passing a `comparer` to the `ToDictionary` call when it's created.

Reply:
I can, just want to notice that in ticket is written to add case insensitive support only for DataSource options.

Reply:
That's because `DataSource` is the only keyword supported by the connection string at the moment. In general, all keywords in the connection string are case insensitive: [Connection string syntax](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings#connection-string-syntax)